src/main/java/segelzwerg/sporttooolbox/IUnits/Distance.java

```diff
@@ -13,6 +13,14 @@ public Distance(float kilometer, float meter) {
 		this.meter = meter % 1000;
 	}
 
+
+    public Distance addDistance(Distance toAdd)
+    {
+        float kilometer= this.kilometer + toAdd.kilometer + (int) (this.meter+toAdd.meter) / 1000;
+        float meter= (this.meter+toAdd.meter) % 1000;
```

Comment:
Can you refactor it, so there is no duplication with the constructor meaning an extracting a method from it and replacing it in this method and the constructor. Or we could also just add the values and do the overflow in the constructor.

Reply:
done, for now i thought let constructor handle it.